internal/server/middleware.go

```diff
@@ -97,15 +102,22 @@ func getDB(c *gin.Context) *gorm.DB {
 func authenticatedMiddleware() gin.HandlerFunc {
 	return func(c *gin.Context) {
 		db := getDB(c)
-		authnUser, err := requireAccessKey(db, c.Request)
+		authned, err := requireAccessKey(db, c.Request)
 		if err != nil {
 			sendAPIError(c, err)
 			return
 		}
 
-		// TODO: save authnUser in a single key
-		c.Set("key", authnUser.AccessKey)
-		c.Set("identity", authnUser.User)
+		rCtx := RequestContext{
+			Request:       c.Request,
+			DBTxn:         db,
+			Authenticated: authned,
+		}
+		c.Set("requestContext", rCtx)
```

Comment:
we should change all the existing middleware and handlers to use our own context. I assume that's coming later.

Reply:
Ya, that's going to be a big change. I've got a few next steps in #2716, and #2717